pkg/tasks/metricsserver.go

```diff
@@ -155,7 +155,32 @@ func (t *MetricsServerTask) create(ctx context.Context) error {
 			}
 		}
 
-		dep, err := t.factory.MetricsServerDeployment(cacm, scas, mcs)
+		tlsSecret, err := t.client.WaitForSecretByNsName(ctx, types.NamespacedName{Namespace: t.namespace, Name: "metrics-server-tls"})
+		if err != nil {
+			return fmt.Errorf("failed to wait for metrics-server-tls secret: %w", err)
+		}
+
+		apiAuthConfigmap, err := t.client.WaitForConfigMapByNsName(ctx, types.NamespacedName{Namespace: "kube-system", Name: "extension-apiserver-authentication"})
+		if err != nil {
+			return fmt.Errorf("failed to wait for kube-system/extension-apiserver-authentication configmap: %w", err)
+		}
+
+		secret, err := t.factory.MetricsServerSecret(tlsSecret, apiAuthConfigmap)
+		if err != nil {
+			return fmt.Errorf("failed to create metrics-server secret: %w", err)
+		}
+
+		err = t.deleteOldMetricsServerSecrets(secret.Labels["monitoring.openshift.io/hash"])
```

Comment:
Not related to the PR: Couldn't this be replaced by a hash on the Deployment itself as we do for the certs?
So we have only one way of "reloading the Deployment if the secret changes".
(I'd create a ticket for that)
(cc @simonpasquier in case you know that this logic is needed for other things than reloading)

Reply:
I wasn't sure either why this was there for prometheus-adapter as well. Happy to move to similar logic we have for other secrets if this is not needed

Reply:
+1 on doing that since we [already use the other logic within MS](https://github.com/openshift/cluster-monitoring-operator/blob/f0dea72dd9e387f0239410a3d17a5fb2ca4a6507/pkg/manifests/manifests.go#L2057-L2069).